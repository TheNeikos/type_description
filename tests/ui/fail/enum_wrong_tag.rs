//
//   This Source Code Form is subject to the terms of the Mozilla Public
//   License, v. 2.0. If a copy of the MPL was not distributed with this
//   file, You can obtain one at http://mozilla.org/MPL/2.0/.
//

#![allow(dead_code)]

use type_description::TypeDescription;

#[derive(Debug, TypeDescription)]
#[description(foobar)]
enum Door {
    Wooden,
    Steel,
    Plastic,
}

fn main() {
    // Empty
}
