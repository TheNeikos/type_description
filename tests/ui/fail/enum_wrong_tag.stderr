error: Enums need to specify what kind of tagging they use

         = help: Use #[description(untagged)] for untagged enums, and #[description(tag = "type")] for internally tagged variants. Other kinds are not supported.

  --> tests/ui/fail/enum_wrong_tag.rs:13:6
   |
13 | enum Door {
   |      ^^^^
